### ISSUE TITLE
Client isn't really very Go like

### ISSUE THREAD

**Issue by crzyuk** [MarketDataApp/sdk-go#1]
Hi,

Having tried the client I have to say this isn't really written that well and it's not that nice to use. 

Firstly you rely on env vars for tokens which is generally unsafe. 

Secondly, the client init is automatically searching for an env var so you get an error even if you attempt to use the NewClient function with a token fetch from a secure source. 

Thirdly the client is some strange global variable which isn't that nice to be honest. If you're creating a client with a token as above you end up with a strange floating variable that you have to do something with to avoid linting errors and trying to test things with a hidden global variably is virtually impossible..

Not sure what the aim was here?

Don't get me wrong, the API is great but this client isn't that great from a Go point of view.

**Comment by MarketDataApp** [comment]
Thanks a lot for the feedback. This was my first attempt to make something in Go, so I appreciate the suggestions.

I'll try to put some more work into this next week to address the issues you brought up.

Feel free to make any other suggestions as well and I will try to take them into account.

**Comment by crzyuk** [comment]
No worries at all. Thanks for the reply.

If I have time over the next week I'll try and submit a PR to make it a bit more usable - if that sounds ok?

**Comment by MarketDataApp** [comment]
Great, sounds great. I'll wait on your PR before making any further fixes/updates then, so we're not overlapping one another.

**Comment by MarketDataApp** [comment]
Thanks for this — it was a fair critique, and v2 is the answer to it. `v2.0.0` is now released: a complete rewrite rather than a patch.

Taking your three points in order.

**Env vars for tokens.** The token is now an explicit client option. The environment is only the last resort in a documented cascade (`.env` < env vars < client options < method params), and you can opt out of `.env` entirely with `WithoutDotEnv()`.

**Init erroring when you supply your own token.** Fixed. `NewClient` takes the token you give it and never consults the environment when you do. There are no `init()` side effects and no network calls on import. This runs with `MARKETDATA_TOKEN` unset:

```go
c, err := marketdata.NewClient(marketdata.WithToken(secretFromYourVault))
if err != nil {
    return err
}
defer c.Close()

quote, _, err := c.Stocks.Quote(ctx, "AAPL")
```

**The global variable.** Gone. `NewClient` returns a `*marketdata.Client` that you own, pass, and close. Nothing is hidden in package state, so tests construct their own client against an `httptest.Server` with no global to reset between cases.

A few other things that follow from the same rewrite: every method takes a `context.Context` first, errors are typed for `errors.Is`/`errors.As`, and the module has no third-party runtime dependencies.

```bash
go get github.com/MarketDataApp/sdk-go/v2
```

v1 is untouched — the `v1.2.0` tag still resolves, and the code stays browsable on the `v1` branch. See [docs/MIGRATION.md](https://github.com/MarketDataApp/sdk-go/blob/main/docs/MIGRATION.md) if you want to move across.

Closing this as resolved by v2. Please do reopen or file a new issue if the new API still gets in your way.
